libvast/src/system/node.cpp

```diff
@@ -27,192 +29,207 @@
 #include "vast/concept/printable/stream.hpp"
 #include "vast/concept/printable/to_string.hpp"
 #include "vast/concept/printable/vast/json.hpp"
+#include "vast/detail/assert.hpp"
 #include "vast/json.hpp"
 #include "vast/logger.hpp"
 
 #include "vast/system/accountant.hpp"
 #include "vast/system/consensus.hpp"
-#include "vast/system/node.hpp"
 #include "vast/system/spawn.hpp"
 
-using namespace std::string_literals;
+using std::string;
+
 using namespace caf;
 
-namespace vast {
-namespace system {
+namespace vast::system {
 
 namespace {
 
-using node_ptr = stateful_actor<node_state>*;
+// Local commands need access to the node actor.
+thread_local node_actor* this_node;
```

Comment:
Have you thought about storing the node actor in the registry, as we do for the accountant? I wonder what motivated the thread-local variable here.

Reply:
> The underlying issue is the interface of command that doesn't allow passing in extra arguments. We would need a generic way of doing so, though.

Ah, that's the reason. Thanks for elaborating.